### PR TITLE
added feature for creating schema

### PR DESCRIPTION
**Pull request by NoRage28**

### CODE REVIEW

**Review by bakarovroman — CHANGES_REQUESTED** [review]
one pull request, one migrations file

File: fake_service/dataset/tests.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
+# Create your tests here.
```
Comment (bakarovroman) [inline fake_service/dataset/tests.py:3]: tests?

File: fake_service/dataset/views.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import viewsets
+
+from .models import Schema
+from .serializers import CreateSchemaSerializer
+
+
+class SchemaViewSet(viewsets.ModelViewSet):
+    queryset = Schema.objects.all()
+    serializer_class = CreateSchemaSerializer
```
Comment (bakarovroman) [inline fake_service/dataset/views.py:9]: use 
```
    def get_serializer_class(self):
        serializers = {
            "create": CreateSchemaSerializer,
        }
        return serializers.get(self.action)
```

File: fake_service/dataset/serializers.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    def create(self, validated_data):
+        return Schema.objects.create(**validated_data)
+
+    def update(self, instance, validated_data):
+        instance.title = validated_data.get('title', instance.title)
+        instance.column_separator = validated_data.get('column_separator', instance.column_separator)
+        instance.column_character = validated_data.get('column_character', instance.column_character)
+        instance.columns = validated_data.get('columns', instance.columns)
+        instance.save()
+        return instance
```
Comment (250kgBeast) [inline fake_service/dataset/serializers.py:53]: I think [model serializer](https://www.django-rest-framework.org/api-guide/serializers/#modelserializer) may be used here

File: fake_service/dataset/tests.py (hunk 1 above)
Comment (250kgBeast) [inline fake_service/dataset/tests.py:3]: tests?

File: fake_service/dataset/serializers.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    name = serializers.CharField()
+    type = serializers.ChoiceField(choices=SCHEMA_COLUMN_TYPE)
+    value_from = serializers.IntegerField(min_value=0, required=False)
+    value_to = serializers.IntegerField(min_value=1, required=False)
+
+    class Meta:
+        validators = [ValueFieldsValidator()]
+
+
+class CreateSchemaSerializer(serializers.ModelSerializer):
```
Comment (250kgBeast) [inline fake_service/dataset/serializers.py:41]: rename to SchemaSerializer

File: fake_service/dataset/tests/test_schema.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        url = reverse('schema-list')
+        response = self.client.get(path=url)
+        result = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIsInstance(result, list)
+        self.assertEqual(result[0]['title'], 'Schema_1')
+        self.assertEqual(result[0]['column_separator'], '.')
+        self.assertEqual(result[0]['column_character'], '"')
+        self.assertEqual(result[0]['columns'], self.columns_data)
```
Comment (250kgBeast) [inline fake_service/dataset/tests/test_schema.py:28]: for key, value in result.items():
   self.assertEqual(getattr(self.schema, key), value)

File: fake_service/dataset/tests/test_schema.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                'column_character': '\'',
+                'columns': [{"name": "Column_1",
+                             "type": "integer",
+                             "value_from": "1",
+                             "value_to": "20"}]}
+        response = self.client.post(path=url, data=data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(Schema.objects.all().count(), 2)
+        self.assertEqual(Schema.objects.last().title, 'Schema_2')
```
Comment (250kgBeast) [inline fake_service/dataset/tests/test_schema.py:43]: created_schema = Schema.objects.last()
for key, value in data.items():
    self.assertEqual(getattr(created_schema, key), value)

**Pull request merged**